pkg/policy/resolve.go

```diff
@@ -164,6 +162,34 @@ func (p *EndpointPolicy) Detach() {
 	p.selectorPolicy.removeUser(p)
 }
 
+func NewMapStateWithInsert() (MapState, func(k Key, e MapStateEntry)) {
```

Comment:
I plan to remove this function in a later change, but added this for now:

> // NewMapStateWithInsert returns a new MapState and an insert function that can be used to populate
// it. We keep general insert functions private so that the caller can only insert to this specific
// map.

Reply:
thanks 🙏